blazor/diagram/nodes/nodes.md

```diff
@@ -258,7 +258,63 @@ public void RemoveNodes()
     nodes.RemoveAt(0);
 }
 ```
+### How to clone the node at runtime
+[Clone](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.Diagram.Node.html#Syncfusion_Blazor_Diagram_Node_Clone) is a virtual method of the node that is used to create a copy of a diagram object. After cloning, we need to set the ID for cloned nodes. The following code demonstrates how to clone the nodes during runtime.
 
+```cshtml
+@using Syncfusion.Blazor.Diagram
+@using System.Collections.ObjectModel
+@inject IJSRuntime js
+<input type="button" id="add" value="Clone Node" @onclick="@CloneNode" />
+<SfDiagramComponent @ref="diagram" Width="50%" Height="500px" @bind-Nodes="NodeCollection"></SfDiagramComponent>
+@functions
+{
+
+    SfDiagramComponent diagram;
+    public DiagramObjectCollection<Node> NodeCollection = new DiagramObjectCollection<Node>();
+    protected override void OnInitialized()
+    {
+        Node node1 = new Node()
+            {
+                ID = "node1",
+                OffsetX = 100,
+                OffsetY = 200,
+                Height = 100,
+                Width = 100,
+                Annotations = new DiagramObjectCollection<ShapeAnnotation>()
+                    {
+                        new ShapeAnnotation()
+                        {
+                            Content = "node1"
+                        },
+                    },
+                Shape = new FlowShape() { Type = NodeShapes.Flow, Shape = NodeFlowShapes.Process }
+            };
+        NodeCollection.Add(node1);
+    }
+    public async Task CloneNode()
+    {
+        Node node = NodeCollection[0].Clone() as Node;
+        node.ID = RandomId();
+        node.OffsetX += 25;
+        node.OffsetY += 25;
+        await diagram.AddDiagramElements(new DiagramObjectCollection<NodeBase>() { node });
+    }
+
+    internal string RandomId()
+    {
+        Random random = new Random();
+        const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXTZabcdefghiklmnopqrstuvwxyz";
+#pragma warning disable CA5394 // Do not use insecure randomness
+        return new string(Enumerable.Repeat(chars, 5)
+          .Select(s => s[random.Next(s.Length)]).ToArray());
+#pragma warning restore CA5394 // Do not use insecure randomness
+    }
+
+}
+```
+You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/Blazor-Diagram-Examples/tree/master/UG-Samples/Nodes/ActionsofNodes)
```

Comment:
You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/Blazor-Diagram-Examples/tree/master/UG-Samples/Nodes/ActionsofNodes).